src/main/kotlin/io/github/inductiveautomation/kindling/MainPanel.kt

```diff
@@ -139,59 +148,91 @@ class MainPanel : JPanel(MigLayout("ins 6, fill")) {
-                                openFiles(selectedFiles, tool)
-                            }
-                        },
-                    )
+    private val fileMenu = JMenu("File").apply {
+        add(openAction)
+        for (tool in Tool.tools) {
+            add(
+                Action(
+                    name = "Open ${tool.title}",
+                ) {
+                    fileChooser.fileFilter = tool.filter
+                    fileChooser.chooseFiles(this@MainPanel)?.let { selectedFiles ->
+                        openFiles(selectedFiles, tool)
+                    }
+                },
+            )
+        }
+        if (!SystemInfo.isMacOS) {
+            addSeparator()
+            add(
+                Action("Preferences") {
+                    preferencesEditor.isVisible = true
+                    preferencesEditor.toFront()
                 }
-            },
-        )
+            )
+        }
+    }
+
+    private val pasteMenu = JMenu("Paste").apply {
+        for (clipboardTool in Tool.tools.filterIsInstance<ClipboardTool>()) {
+            add(
+                Action(
+                    name = "Paste ${clipboardTool.title}",
+                ) {
+                    val clipboard = Toolkit.getDefaultToolkit().systemClipboard
+                    if (clipboard.isDataFlavorAvailable(DataFlavor.stringFlavor)) {
+                        val clipString = clipboard.getData(DataFlavor.stringFlavor) as String
+                        openOrError(clipboardTool.title, "clipboard data") {
+                            clipboardTool.open(clipString)
+                        }
+                    } else {
+                        LOGGER.info("No string data found on clipboard")
+                    }
+                },
+            )
+        }
+    }
+
+    private val menuBar = JMenuBar().apply {
+        add(fileMenu)
+        add(pasteMenu)
         add(
-            JMenu("Paste").apply {
-                for (clipboardTool in Tool.tools.filterIsInstance<ClipboardTool>()) {
-                    add(
-                        Action(
-                            name = "Paste ${clipboardTool.title}",
-                        ) {
-                            val clipboard = Toolkit.getDefaultToolkit().systemClipboard
-                            if (clipboard.isDataFlavorAvailable(DataFlavor.stringFlavor)) {
-                                val clipString = clipboard.getData(DataFlavor.stringFlavor) as String
-                                openOrError(clipboardTool.title, "clipboard data") {
-                                    clipboardTool.open(clipString)
-                                }
-                            } else {
-                                LOGGER.info("No string data found on clipboard")
-                            }
-                        },
-                    )
+            JMenu("Help").apply {
+                add(debugMenu)
+                add(Action("Forum") {
```

Comment:
Slightly goofy, but this way there's never an empty Help menu with any combination of platforms/settings :tap-forehead: